assayist/processor/main_analyzer.py

```diff
@@ -0,0 +1,135 @@
+        """
+        Read the build info and contruct the Component.
+
+        Returns: (Component, canonical_version)
+
+        """
+        if build_type == 'build':  # rpm build
+            cnamespace = 'redhat'
+            cname = build_info['name']
+            ctype = 'rpm'
+            cversion = '%s-%s' % (build_info['version'], build_info['release'])
+        elif build_type == 'maven':
+            # What we really want is the contents of the "Maven groupId" etc fields.
+            # However they don't seem to be included in the brew response?!
+            # Instead let's parse it out with what is (as best as I can tell) the algorithm.
+            cnamespace, cname = build_info['name'].split('-', 1)
+            cversion = build_info['version'].replace('_', '-')
+            ctype = 'java'
+        elif build_type == 'buildContainer':
+            cnamespace = 'docker-image'
+            cname = build_info['name']
+            cversion = '%s-%s' % (build_info['version'], build_info['release'])
+            ctype = 'image'
+        else:
+            return None, None
+
+        component = self.get_or_create_component(
+            canonical_namespace=cnamespace,
+            canonical_name=cname,
+            canonical_type=ctype)
+        return component, cversion
+
+    def run(self):
+        """Do the actual processing."""
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        task_info = self.read_metadata_file(self.TASK_FILE)
+
+        build_type = None
+        if task_info:
+            build_type = task_info['method']
+
+        # construct the component
+        component, canonical_version = self.construct_and_save_component(build_type, build_info)
+
+        # construct the SourceLocation
+        source = build_info['source']
+        sourceLocation = self.get_or_create_source_location(source, canonical_version)
+        sourceLocation.component.connect(component)
+
+        # construct the build object
+        build = self.get_or_create_build(build_info['id'], build_type)
+        build.source_location.connect(sourceLocation)
+
+        # if it's an rpm build look at the rpm output and create artifacts
+        if build_type == 'build':
+            rpms_info = self.read_metadata_file(self.RPM_FILE)
+            for rpm_info in rpms_info:
+                buildroot_id = rpm_info['buildroot_id']
+                rpm = self.get_or_create_rpm_artifact_from_hash(rpm_info)
+                rpm.build.connect(build)
+                self.map_buildroot_to_artifact(buildroot_id, rpm)
+
+            self.read_and_save_buildroots()
+            return  # finished processing, rpm builds don't have anything else
+
+        # else not an rpm build, record the artifacts
+        archives_info = self.read_metadata_file(self.ARCHIVE_FILE)
+        images_rpm_info = self.read_metadata_file(self.IMAGE_RPM_FILE)
+        for archive_info in archives_info:
```

Comment:
We should check `archive_info["type"] == "docker-image"`, so we skip over logs.

Reply:
@twaugh Ideally this would work with other build types (like maven builds), so as it stands I'm NACKing this idea. Can you give me an example of a build that lists logs in the archives section? As far as I've seen they're always in a separate "logs" listing and wouldn't be included here.

Reply:
Any container image build will include archives where archive["type"] is "log".